File: C++/reverse_words_in_a_string_III.cpp
#include <iostream>
#include <map>  // map is in sorted order
#include "./reverse_words_in_a_string_III.h"
#include "./utility.h"

using namespace std;

string Solution::reverseWords(string s) {
    string output = "";
    return output;
}

int main() {
    string s = "Let's take LeetCode contest";

    Solution* sol = new Solution();
    string output = sol->reverseWords(s);
    delete sol;

    cout << output << "\n";

    return 0;
}

File: Python2/reverse_words_in_a_string_III.py
class Solution(object):
    def reverseWords(self, s):
        # total: O(n+n+n) => O(n)
        words = s.split()  # O(n)
        return ' '.join([w[::-1] for w in words]) # O(n + n)



if __name__ == '__main__':
    s = "Let's take LeetCode contest"
    sol = Solution()
    print sol.reverseWords(s)